python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1163,6 +1173,12 @@ def create_run_for_job(
             parent_job_snapshot=job_def.get_parent_job_snapshot(),
             external_job_origin=external_job_origin,
             job_code_origin=job_code_origin,
+            external_partitions_definition_data=external_partitions_definition_from_def(
+                job_def.partitions_def
+            )
+            if job_def.partitions_def
+            and can_build_external_partitions_definition_from_def(job_def.partitions_def)
```

Comment:
Without this check, we would just error, right? What's wrong with that? Might be better to find out early than discover a problem later along the way?

Reply:
This check guards against the backcompat dynamic partitions case (a function that returns partition keys)

Without it we raise an error, but what we want is to not yield the asset materialization planned events

Reply:
> This check guards against the backcompat dynamic partitions case (a function that returns partition keys)

Is it currently possible to launch a range backfill using function-based dynamic partitions? I would be comfortable saying that's not allowed.

Reply:
It's not possible from the UI but I don't think we gate it if you were to yield a run request with tags or something like that.

Agree that we can make it not allowed, I don't think anyone is doing that anyway